src/MICore/CommandFactories/gdb.cs

```diff
@@ -295,5 +295,46 @@ public override async Task<string[]> AutoComplete(string command, int threadId,
 
             return matchlist?.AsStrings;
         }
+
+        public override IEnumerable<Guid> GetSupportedExceptionCategories()
+        {
+            const string CppExceptionCategoryString = "{3A12D0B7-C26C-11D0-B442-00A0244A1DD2}";
+            return new Guid[] { new Guid(CppExceptionCategoryString) };
+        }
+
+        public override async Task<IEnumerable<ulong>> SetExceptionBreakpoints(Guid exceptionCategory, IEnumerable<string> exceptionNames, ExceptionBreakpointStates exceptionBreakpointStates)
+        {
+            string command;
+            Results result;
+            List<ulong> breakpointNumbers = new List<ulong>();
+
+            if (exceptionNames == null) // set breakpoint for all exceptions in exceptionCategory
+            {
+                command = string.Format(CultureInfo.InvariantCulture, "-catch-throw");
+                result = await _debugger.CmdAsync(command, ResultClass.done);
+                var breakpointNumber = result.Find("bkpt").FindUint("number");
+                breakpointNumbers.Add(breakpointNumber);
+            }
+            else // set breakpoint for each exceptionName in exceptionNames
+            {
+                command = string.Format(CultureInfo.InvariantCulture, "-catch-throw -r ");
+                foreach (string exceptionName in exceptionNames)
+                {
+                    result = await _debugger.CmdAsync(command + exceptionName, ResultClass.done);
```

Comment:
>exceptionName [](http://example.com/codeflow?start=64&length=13)

Do we need to do anything to prevent partial string matches here? For example, suppose the debugger is configured to break on `Foo` but the exception raised is a `NFoo` or `FooN`. Do we need to surrond `exceptionName` with `\b` or something like that?

Reply:
Thank you for bringing this up. Fixed using `^` to match the start of the string and `$` to match the end of the string. Source: https://stackoverflow.com/questions/4581412/how-to-make-regexs-only-match-on-exact-matches/4581431

Reply:
You might want `\b` (or similar if `\b` isn't supported) instead of `^` and `$`, but you will need to test to find out: what happens if you raise an exception of type `const Foo` or `Foo*`?

Reply:
Using `^(exceptionName)$` with

`const Foo`
![image](https://user-images.githubusercontent.com/60245970/143058851-4b547f2a-ae79-4f34-ac80-a5c7efb6644d.png)

but the catchpoint does not hit when an exception of type `Foo*` is raised.

Reply:
With `\b(exceptionName)\b`, however, the catchpoint hits in both test scenarios.

`const Foo`
![image](https://user-images.githubusercontent.com/60245970/143074339-beb84b3c-7c35-461f-9eff-77b2cc942142.png)

`Foo*`
![image](https://user-images.githubusercontent.com/60245970/143073617-125ac3ee-bac6-444f-871d-af8301f99829.png)

Reply:
Excellent. So you want `\b` then.